RevenueCat/Example/PurchasesListener.cs

```diff
@@ -74,99 +161,404 @@ private void SwitchUser()
+                                        }
+                                        else
+                                        {
+                                            if (cancelled)
+                                            {
+                                                infoLabel.text = "purchase cancelled!";
+                                            }
+                                            else
+                                            {
+                                                infoLabel.text +=
+                                                    $"Purchase of {identifier} successful!\ncustomerInfo:\n{customerInfo}";
+                                            }
+                                        }
+                                    });
+                            }
+                        }
+                    });
             }
         });
     }
 
-    void RestoreClicked()
+    void BuyFirstPackageWithDiscount()
     {
         var purchases = GetComponent<Purchases>();
-        purchases.RestorePurchases((customerInfo, error) =>
+        purchases.GetOfferings((offerings, error) =>
         {
             if (error != null)
             {
                 LogError(error);
             }
             else
             {
-                DisplayCustomerInfo(customerInfo);
+                var packages = GetPackages(offerings)
+                    .Where(package => package.StoreProduct.Discounts is { Length: > 1 }) // map to product ids
+                    .ToArray();
+
+                if (packages.Length == 0)
+                {
+                    infoLabel.text = "No promotional offers available for any products";
+                    return;
+                }
+
+                infoLabel.text = "";
+                var package = packages.First();
+                purchases.GetPromotionalOffer(package.StoreProduct, package.StoreProduct.Discounts.First(),
+                    (promoOffer, promoOfferError) =>
+                    {
+                        lock (this)
+                        {
+                            if (promoOfferError != null)
+                            {
+                                LogError(promoOfferError);
+                            }
+                            else
+                            {
+                                purchases.PurchaseDiscountedProduct(package.StoreProduct.Identifier, promoOffer,
+                                    (identifier, customerInfo, cancelled, purchaseError) =>
+                                    {
+                                        if (purchaseError != null)
+                                        {
+                                            LogError(purchaseError);
+                                        }
+                                        else
+                                        {
+                                            if (cancelled)

```

Comment:
Isn't purchaseError not null here? I think it will be a PurchaseCancelledError but I'm not sure

Reply:
you're right! updated.